example/scalalib/native/1-simple/src/Foo.scala

```diff
@@ -0,0 +1,25 @@
+package foo
+
+import scala.scalanative.libc._
+import scala.scalanative.unsafe._
+import mainargs.{main, ParserForMethods}
+
+object Foo {
+  
+  def generateHtml(text: String): CString = {
+    val html = "<h1>" + text + "</h1>\n"
+
+    implicit val z: Zone = Zone.open()
+    val cResult = toCString(html)
+    cResult
```

Comment:
Doesn't `cResult` escape here the `Zone` where it was allocated?

Reply:
@c0d33ngr please open a new PR with these changes, it's easier to discuss individual code changes on the PR, and when we're done we can just click the merge button :)